src/atldld/cli.py

```diff
@@ -57,5 +60,76 @@ def cache_folder():
 info.add_command(version)
 info.add_command(cache_folder)
 
+
+# ============================= Dataset subcommand =============================
+
+
+@click.group(help="Commands related to atlas datasets")
+def dataset():
+    """Run dataset subcommands."""
+
+
+@click.command("info", help="Get information for a given dataset ID")
+@click.argument("dataset_id", type=int)
+def dataset_info(dataset_id):
+    """Get information for a given dataset ID."""
+    import textwrap
+
+    import atldld.dataset
+    from atldld import requests
+
+    # Send request
+    rma_parameters = requests.RMAParameters(
+        "SectionDataSet",
+        criteria={"id": dataset_id},
+        include=("genes", "section_images"),
+    )
+    try:
+        msg = requests.rma_all(rma_parameters)
+    except requests.RMAError as exc:
+        click.secho(
+            f"An error occurred while querying the AIBS servers: {str(exc)}",
+            fg="red",
+        )
+
+    # Check response
+    if len(msg) == 0:
+        click.secho(f"Dataset with ID {dataset_id} does not exist", fg="red")
+        raise click.Abort
+    elif len(msg) > 1:
+        click.secho("Something went wrong: got more than one dataset", fg="red")
+        raise click.Abort
+
+    # Print response
+    meta = msg[0]
+    section_images = meta.pop("section_images")
+    r_str = meta["red_channel"] or "-"
+    g_str = meta["green_channel"] or "-"
+    b_str = meta["blue_channel"] or "-"
+    plane_of_section = atldld.dataset.PlaneOfSection(meta["plane_of_section_id"])
+    reference_space = atldld.dataset.ReferenceSpace(meta["reference_space_id"])
+    output = f"""
+    ID                       : {meta["id"]}
+    Sphinx ID                : {meta["sphinx_id"]}
+    Specimen ID              : {meta["specimen_id"]}
+    Name                     : {meta["name"] or "-"}
+    Failed                   : {"Yes" if meta["failed"] else "No"}
+    Expression               : {"Yes" if meta["expression"] else "No"}
+    Gene(s)                  : {", ".join(gene["acronym"] for gene in meta["genes"])}
+    RGB channels             : {r_str} / {g_str} / {b_str}
+    Section thickness        : {meta["section_thickness"]}µm
+    Plane of section         : {plane_of_section}
+    Number of section images : {len(section_images)}
+    Reference space          : {reference_space.value} ({reference_space})
+    """
+    click.secho(textwrap.dedent(output).strip(), fg="green")
+
+
+root.add_command(dataset)
+dataset.add_command(dataset_info)
```

Comment:
Just out of curiosity, why did you decide to go for this syntax?

AFAIK one can equivalently do

```python
@click.group(help="Commands related to atlas datasets")
def dataset():
    """Run dataset subcommands."""
    ...

@dataset.command("info", help="Get information for a given dataset ID")
@click.argument("dataset_id", type=int)
def dataset_info(dataset_id):
    ...
```

Reply:
Like I mentioned during our call, I totally forgot it was possible that way, so no particular reason :)

I just checked the [click docs on nesting commands](https://click.palletsprojects.com/en/8.0.x/quickstart/#nesting-commands) and they say things like

> For simple scripts, it’s also possible to automatically attach and create a command by using the Group.command() decorator instead.

and

> Instead of using the @group.command() decorator, commands can be decorated with the plain @click.command() decorator and registered with a group later with group.add_command(). This could be used to split commands into multiple Python modules.

This sounds to me as though they're suggesting using the syntax you suggest for simple CLIs and `group.add_command` for more complicated setups, with commands potentiall spread across modules.

I don't really have much experience with click, so I'm not sure I can fully assess the situation.

Do you have any suggestions?

Reply:
Interesting:) We can keep it the way it is:)

Reply:
(Can be resolved)